src/api/java/baritone/api/utils/RotationUtils.java

```diff
@@ -215,9 +216,10 @@ public static Optional<Rotation> reachable(EntityPlayerSP entity, BlockPos pos,
      * @param blockReachDistance The block reach distance of the entity
      * @return The optional rotation
      */
-    public static Optional<Rotation> reachableOffset(Entity entity, BlockPos pos, Vec3d offsetPos, double blockReachDistance, boolean wouldSneak) {
+    public static Optional<Rotation> reachableOffset(EntityPlayerSP entity, BlockPos pos, Vec3d offsetPos, double blockReachDistance, boolean wouldSneak) {
+        IPlayerContext ctx = BaritoneAPI.getProvider().getBaritoneForPlayer(entity).getPlayerContext();
```

Comment:
Technically this is api breaking, would be surprised if anyone calls it on arbitrary entities though.

Does this throw an NPE if I call it on a fake player?

Reply:
> Technically this is api breaking, would be surprised if anyone calls it on arbitrary entities though.

Yeah I doubt that anybody is actually calling this. Either way I'm going to add back the one with the `Entity` parameter as `@Deprecated` and change the updated version to take an `IPlayerContext` instead of resolving the baritone instance from the player to get it.

> Does this throw an NPE if I call it on a fake player?

Yes. It'd only work correctly for `mc.player` on the main branch.